.github/workflows/link-check.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and check Links
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build_and_serve:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install Dependencies
+        run: yarn install
+
+      - name: Build Next.js App
+        run: yarn build
+
+      - name: Serve App Locally
+        run: yarn start &
+
+      - name: Wait for App to Start
+        run: sleep 20
+
+  linkChecker:
+    runs-on: ubuntu-latest
+    needs: build_and_serve
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Link Checker
+        uses: lycheeverse/lychee-action@v1
+        with:
+          fail: true
+          args: '--base https://json-schema.org --verbose --no-progress **/*.md' 
```

Comment:
Hey, 
To address @Relequestual 's feedback, we should revert my suggestion of setting base as the deployed build. As, deployed build with not include changes introduced in a PR.

> No, we DO want it to be on pull request. We want to avoid deploying with a broken link.
We need to work out how to also check the deployable version in the PR.
It might be a case of running the build before link checking, then doing a remap as part of the config to address localhost.